src/app/ui-components/button/button.component.ts

```diff
@@ -0,0 +1,15 @@
+import { Component, Input } from '@angular/core';
+import { NgClass } from '@angular/common';
+
+@Component({
+  selector: 'button[alg-button], a[alg-button]',
+  templateUrl: './button.component.html',
+  styleUrls: [ './button.component.scss' ],
+  standalone: true,
+  imports: [
+    NgClass
+  ]
+})
+export class ButtonComponent {
+  @Input() icon?: string;
```

Comment:
Shouldn't mark your new UI component as `changeDetection: ChangeDetectionStrategy.OnPush` ?

And (at least for the next UI components that you create), shouldn't you use signals directly (looks easier (handle the "required") and prevent loosing context when we'll apply an automated conversion)